pmd-core/src/main/java/net/sourceforge/pmd/RuleSetReferenceId.java

```diff
@@ -85,12 +76,21 @@
 @Deprecated
 @InternalApi
 public class RuleSetReferenceId {
+
+    // todo this class has issues... What is even an "external" ruleset?
+    //  terminology and API should be clarified.
```

Comment:
Keep in mind at some point we marked it as InternalApi.
Clarification should happen anyway of course...